python/pylibcudf/pylibcudf/table.pyx

```diff
@@ -100,8 +121,10 @@ cdef class Table:
 
             tmp = Table.from_table_view_of_arbitrary(result.tbl.get().view(), result)
             self._columns = tmp.columns()
+        elif hasattr(arrow_like, "__arrow_c_device_stream__"):
+            raise NotImplementedError("Device streams not yet supported")
         elif hasattr(arrow_like, "__arrow_c_array__"):
```

Comment:
Nit: Do we need the `else: raise ValueError("Invalid Arrow object...")` like you do for column?

Reply:
We can. It doesn't really matter because all the possible cases are covered by the types enumerated in the metaclass used for dispatching, so the else clause will be unreachable. I can add it for completeness though.